src/sentry/event_manager.py

```diff
@@ -905,19 +908,51 @@ def get_culprit(data):
     )
 
 
-def _save_aggregate(event, flat_hashes, release, **kwargs):
+def _find_hashes(
+    project,
+    flat_grouphashes,
+    hierarchical_hashes,
+):
+    all_grouphashes = []
+
+    if hierarchical_hashes:
+        hierarchical_grouphashes = {
+            h.hash: h
+            for h in GroupHash.objects.filter(project=project, hash__in=hierarchical_hashes)
+        }
+
+        for hash in reversed(hierarchical_hashes):
+            group_hash = hierarchical_grouphashes.get(hash)
+            if group_hash is None:
+                continue
+
+            all_grouphashes.append(group_hash)
+
+    all_grouphashes.extend(flat_grouphashes)
+
+    for group_hash in all_grouphashes:
+        if group_hash.group_id is not None:
```

Comment:
I think this requires a comment. So I guess the logic here is supposed to be: `group_id` and `group_tombstone_id` can never be set at once and we do in fact want to handle this on every level and not first check for group tombstones and then for groups? I suppose delete and discard will change in how it operates now?

Reply:
So to be clear, I would've expected us to build an entirely separate table or column for the "tombstoning" of hierarchical hashes. I don't think it is a good idea to overload GroupTombstone to also be a marker for "skip this grouphash" depending on whether the hash is hierarchical or not.

Given that I don't think delete and discard will change at all.

Reply:
armin clarified, I will add a comment